scylla-cql/src/frame/value.rs

```diff
@@ -886,9 +886,13 @@ impl Value for BigDecimal {
         let (value, scale) = self.as_bigint_and_exponent();
 
         let serialized = value.to_signed_bytes_be();
-        let serialized_len: i32 = serialized.len().try_into().map_err(|_| ValueTooBig)?;
+        let serialized_len: i32 = serialized
+            .len()
+            .checked_add(4)
+            .ok_or(ValueTooBig)
+            .and_then(|x| x.try_into().map_err(|_| ValueTooBig))?;
```

Comment:
Now that I think about it, I'm not sure whether it is really possible to trigger the issue that supposedly happens here. It would require `serialize` to take up nearly all or more than available memory in the system (i.e. larger than `usize::MAX - 4).

I'll leave up the decision up to you what to do with it - another option would be to document why it's fine to add 4 and not check for overflow.

Reply:
I think it's better to leave the check anyway.